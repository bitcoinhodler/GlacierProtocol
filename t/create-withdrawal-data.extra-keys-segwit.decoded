{
  "txid": "e44c7c3ba5be5063f76aefd24d08624777addcf41530582cabcd0a0480b8b7dc",
  "hash": "d1f457785f111c7c0a1db45d8125a3eb8b925f20698aae10070f092b1056acf9",
  "version": 2,
  "size": 406,
  "vsize": 190,
  "weight": 760,
  "locktime": 0,
  "vin": [
    {
      "txid": "2d1591287149560b813706fcd884afc95512fe4638b9e64a14d94bc21c561d4f",
      "vout": 1,
      "scriptSig": {
        "asm": "00201e6d327472a0dbd0cd64ea03229a86cf865f02029a29ad06635c5e6d83213328",
        "hex": "2200201e6d327472a0dbd0cd64ea03229a86cf865f02029a29ad06635c5e6d83213328"
      },
      "txinwitness": [
        "",
        "304402207acc665d5b67286123de79c712853ddf5eb8fffbfe59c36efe45d33364df123402202e654f02bc467f8726142118c052c0b01981ead8a390e52ddd95fb30b4420c5f01",
        "304402200abacd27b04cfb1ed2205dee93b06986b7f16a3e9b116f8140be58b5c0ed914202201dcee485873a1d5a94b3edf974956575bcc280ccf650c633a939d5841d650f9f01",
        "5221027cac2fe4bb16fa9cdd57de980f546d7cddd8704f11dfc37803ce8831955caf3821031344a608e2a2e7f0a4fd3f60e004f633ef4361482b5bbf4e420a07035ca3554121023d6acb654efdd5b6354dd6bfa2e1fab9feab6468d970fdac9b373dfadb11d5952103a6c60cec3e79f91bec8453f534501f05ad2b0b6e6e1faae05e7a2391d65d998054ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.01998100,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 c6e28c80218f62fc1bbd767d53a1d419fd5a7bb0 OP_EQUAL",
        "desc": "addr(2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe)#9fegmkl5",
        "hex": "a914c6e28c80218f62fc1bbd767d53a1d419fd5a7bb087",
        "address": "2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe",
        "type": "scripthash"
      }
    }
  ]
}
