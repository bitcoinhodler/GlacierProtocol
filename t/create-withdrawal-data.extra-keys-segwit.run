#!/bin/bash
set -e

# Test validation of extra keys. Only need 2, providing 4 as I would in the test withdrawal process


$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2Mv6fmwU6Nj6tiSNHzxSDz8vFiEJkYQ46Fm
5221027cac2fe4bb16fa9cdd57de980f546d7cddd8704f11dfc37803ce8831955caf3821031344a608e2a2e7f0a4fd3f60e004f633ef4361482b5bbf4e420a07035ca3554121023d6acb654efdd5b6354dd6bfa2e1fab9feab6468d970fdac9b373dfadb11d5952103a6c60cec3e79f91bec8453f534501f05ad2b0b6e6e1faae05e7a2391d65d998054ae
2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe
1
02000000000101d926a58de53430672cd14cd9e09fd2787921a04d71b1c99577cf97bd716312f00100000017160014764cb845183657d1f080dc2efc03ee0ec76c6603ffffffff02dbfaea000000000017a91481103a6b381860393b0ac7b92cf6c38df88834538780841e000000000017a9141f4a28e4da99092418131207521012b19cd45f3e870247304402201eafedbc2916369246e4b1cf3494f89d915a61ce4ad864291af2caddf204901602205577d825cbcab10975ce4978bf818a7d3b093f5e94f83f7caa03b101dfede9c3012102d98d267ce32007a2c7ba5f83f619f653fce42d0964f9d5de7094078bb27a20c800000000
4
cUY5DRXSmpqT7WpBAkjFz6PPxHoPJ2Cw2VNCa7muihpRjAUJVycs
cQTQEE1ooQHpVi5C4aTcbYPSUMzY2gF1DVdafUsEwnYYaT97UUMy
cQsAGjLXgGXpa48W8kUh3Pci8bMDENoUW5zVkanknDUWd4uxQSPh
cTQHU4ZzwLthJhDU6oxSzmWsTmwhDvFphJCfKFDZxxy6F8j37jRq
10
y

y
INPUT
