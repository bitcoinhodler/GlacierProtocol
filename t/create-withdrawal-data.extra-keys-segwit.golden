Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.02000000 BTC

How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: Key #3: Key #4: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00002220 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00002220, you have 0.01997780 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to 2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.01998100 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.02000000 BTC in unspent supplied transactions
0.01998100 BTC going to destination address 2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe
Fee amount: 0.00001900

Signing with private keys: 
cUY5DRXSmpqT7WpBAkjFz6PPxHoPJ2Cw2VNCa7muihpRjAUJVycs
cQTQEE1ooQHpVi5C4aTcbYPSUMzY2gF1DVdafUsEwnYYaT97UUMy
cQsAGjLXgGXpa48W8kUh3Pci8bMDENoUW5zVkanknDUWd4uxQSPh
cTQHU4ZzwLthJhDU6oxSzmWsTmwhDvFphJCfKFDZxxy6F8j37jRq


Confirm? (y/n): 
Calculating transaction...


Sufficient private keys to execute transaction?
True

Raw signed transaction (hex):
020000000001016fac4e4851a864b8f2621bb031a61e112ca75898fd8f11a3247b7bf0fce07bfc01000000232200201e6d327472a0dbd0cd64ea03229a86cf865f02029a29ad06635c5e6d83213328ffffffff01147d1e000000000017a914c6e28c80218f62fc1bbd767d53a1d419fd5a7bb087040047304402205f6997af0050ecfa13c16f06742cbd2b1dc8e1d158f9f6bd179a226557849d38022030bafdcee05c7a6a2b883ba54178e43262e08e0cc14aba73034040d68081ee760147304402207225db3a0d78fb881aaa6b00d881ff31ab7570be32db01c5fa4d391bef9a8b8102201932fae8e9c6e7c63032425fd48891e660aab7de425f44968eacff418ec3db45018b5221027cac2fe4bb16fa9cdd57de980f546d7cddd8704f11dfc37803ce8831955caf3821031344a608e2a2e7f0a4fd3f60e004f633ef4361482b5bbf4e420a07035ca3554121023d6acb654efdd5b6354dd6bfa2e1fab9feab6468d970fdac9b373dfadb11d5952103a6c60cec3e79f91bec8453f534501f05ad2b0b6e6e1faae05e7a2391d65d998054ae00000000

Transaction fingerprint (md5):
4ca4a1eb6d49e3f1d51ad988deb87370
QR code for transaction written to transaction.png
